### ISSUE TITLE
Default subcommand

### ISSUE THREAD

**Issue by WaffleLapkin** [google/argh#57]
Is it possible to make a default subcommand with `argh`? I want something like this:
```rust
#[derive(FromArgs, PartialEq, Debug)]
/// Top-level command.
struct TopLevel {
    #[argh(subcommand)]
    nested: Sub,
}

#[derive(FromArgs, PartialEq, Debug)]
#[argh(subcommand)]
enum Sub {
    #[argh(default)] // <-----
    One(SubCommandOne),
    Two(SubCommandTwo),
}

/* one and two subcommands*/
``` 
With
- `cmd one ...` resulting in `Sub::One(..)`
- `cmd two ...` resulting in `Sub::Two(..)`
- `cmd ...` resulting in `Sub::One(..)` (the default)

**Comment by mraerino** [comment]
this is a valid way to do this i think:

```rust
/// Service parameters
#[derive(FromArgs, Debug)]
struct Args {
    #[argh(subcommand)]
    command: Option<Command>,
}
```

**Comment by WaffleLapkin** [comment]
@mraerino I don't think this would work. If subcommand doesn't match (i.e.: default should be used) then your code will lose all arguments.

**Comment by mraerino** [comment]
True. You wouldn't be able to use args on your default command. But global args still work.
My expectation is that if you don't specify a subcommand you likely don't need any args specific to it.

**Comment by tjkirch** [comment]
> My expectation is that if you don't specify a subcommand you likely don't need any args specific to it.

FWIW, I did. I was adding subcommands to an existing tool, and wanted one subcommand to be the default so that the tool would function as it did before if no subcommand was given. Its arguments shouldn't be valid for other subcommands, though.
